test/sequential/test-performance.js

```diff
@@ -81,34 +68,41 @@ checkNodeTiming({
   environment: { around: 0 },
   loopStart: -1,
   loopExit: -1
-});
+};
+
+function checkNodeTiming(props) {
+  Object.keys(props).forEach((prop) => {
+    const param = props[prop];
+    const performanceParam = performance.nodeTiming[prop];
+
+    if (param.hasOwnProperty('around')) {
+      assert.strictEqual(typeof performanceParam, 'number');
+      const delta = performanceParam - param.around;
+      assert(Math.abs(delta) < 1000);
+    } else {
+      assert.strictEqual(performanceParam, param,
+                         `mismatch for performance property ${prop}: ` +
+                         `${performanceParam} vs ${param}`);
+    }
+  });
+}
 
-setTimeout(() => {
-  checkNodeTiming({
-    name: 'node',
-    entryType: 'node',
-    startTime: 0,
+checkNodeTiming(timingParams);
+
+setImmediate(() => {
```

Comment:
I would assume there's a reason the original test case used both setImmediate and setTimeout.